lib/portlayer/exec/exec.go

```diff
@@ -171,6 +192,144 @@ func eventCallback(ie events.Event) {
 	return
 }
 
+func waitForContainerStarted(container *Container) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), propertyCollectorTimeout)
+	defer cancel()
+
+	if container.vm == nil {
+		log.Errorf("Container(%s) wait failed for vm is not found", container.ExecConfig.ID)
+		return false
+	}
+	log.Debugf("Waiting for container(%s) started", container.ExecConfig.ID)
+	for _, k := range extraconfig.CalculateKeys(container.ExecConfig, "Sessions.*.Detail.StartTime", "") {
```

Comment:
Having this loop on the outside means that we're calling vSphere once per session - not significant now given numbers of sessions, but will become more significant in the future.